### DISCUSSION TITLE
Need help with Beca thermostat setup

### DISCUSSION THREAD

**Discussion by catdogmaus** [arendst/Tasmota#17641] [Q&A]
Hi, I flashed successfully Beca BHT-6000 thermostat. Blakadder database says that it is confirmed working in Tasmota. I have not been able to find configuration for this device. I haven't used tasmota before so it's possible I've missed something. Any help?

**Comment by sfromis** [comment]
Tasmota does not run on the thermostat, as it is Zigbee. Did you pair it with your Zigbee coordinator running Tasmota?
Did you use the available configuration file? See https://zigbee.blakadder.com/Beca_BHT-6000.html

**Reply by catdogmaus** [reply]
> Tasmota does not run on the thermostat, as it is Zigbee. Did you pair it with your Zigbee coordinator running Tasmota? Did you use the available configuration file? See https://zigbee.blakadder.com/Beca_BHT-6000.html

I had this Tuya device and I was under the impression that Tasmota was suitable for reconfiguring a Tuya device. My plan for zigbee devices is SkyConnect when it becomes available again.

**Reply by sfromis** [reply]
Since you quote Blakadder database "it is confirmed working in Tasmota", you can only have gotten that from the Zigbee repository. Tuya also makes Zigbee devices, albeit often using their own protocols instead of easy interoperability. The Zigbee configuration file linked in the templates page should help with that.

More generally, Tasmota does support many Tuya devices, when they are ESP8266-based. Zigbee devices are a different thing (Tuya or not) and supported in a very different way.

**Reply by catdogmaus** [reply]
> Since you quote Blakadder database "it is confirmed working in Tasmota", you can only have gotten that from the Zigbee repository. Tuya also makes Zigbee devices, albeit often using their own protocols instead of easy interoperability. The Zigbee configuration file linked in the templates page should help with that.
> 
> More generally, Tasmota does support many Tuya devices, when they are ESP8266-based. Zigbee devices are a different thing (Tuya or not) and supported in a very different way.

As I said I had no experience with Tasmota before. When I looked to devices table in blakadder it insists that BHT-6000 should be compatible with ZigbeeHA, Tasmota, ZigbeeToMQTT, deConz, ZigbeeDomoticz, and iobrokerZigbee. So basically with everything. 
I guess I have to repeat just in case, **I don't have and never had** any zigbee devices. This is planned in the future. I converted BHT-6000 as Tuya device to Tasmota. I think the thing I need to figure out now is whether this particular Tuya device can be used in Tasmota or if should I go back to the OEM software?

**Reply by sfromis** [reply]
What do you mean specifically by "converted BHT-6000 as Tuya device to Tasmota"?

The page talks about a Zigbee device where tasmota cannot be installed, unless you modify the hardware significantly. If you have that device, that may be your first Zigbee device. You may have had it paired with a Tuya hub not informing you that it is technically Zigbee, but as I hinted at, the way Tuya uses Zigbee is not great for interoperability anyway, so it is understandable if they do not talk much about Zigbee.

The point of "should be compatible with ZigbeeHA, Tasmota, ZigbeeToMQTT, deConz, ZigbeeDomoticz, and iobrokerZigbee" is that the device has been verified to work with the Zigbee support in those open source projects, when you have a suitable Zigbee coordinator matching those projects.

According to what you found, it should be feasible to make the device work with Tasmota, as long as you have a coordinator running Tasmota. Other coordinators are also reasonable options, of course.

**Reply by catdogmaus** [reply]
>  What do you mean specifically by "converted BHT-6000 as Tuya device to Tasmota"?

If the device works in the Tuya environment, with the Tuya app, and finally I converted it using Tuya Convert, then I am really tempted to assume that it is a Tuya device. I don't have very deep knowledge of various IoT technologies, but I think I have learned  thing that Zigbee device cannot work without zigbee gateway. Also based on this, my bht-6000 working as zigbee would be a real miracle.

**Reply by sfromis** [reply]
If you have been able to use Tuya-convert, it is a different device from what "should be compatible with ZigbeeHA, Tasmota, ZigbeeToMQTT, deConz, ZigbeeDomoticz, and iobrokerZigbee", not a Zigbee device, and the page you looked at is not relevant. As the Tasmota templates repository for non-Zigbee devices ( https://templates.blakadder.com/ ) does not contain any devices named something with "Beca" or "BHT", I know no details. However, it appears that the Beca BHT-6000 exists both in Zigbee and Wifi variants, with the latter plausibly being what you have.

In general, if you have a non-Zigbee Tuya device, it is usually possible to get it to work with Tasmota, but it needs a specific configuration to match the hardware details. If a template had already been created by someone having the device, that would have contained the basics. Here's an example of such a template for another thermostat device. https://templates.blakadder.com/Moes_WHT-HY609.html

You'd likely need something similar, but it may take some experimentation to work out the details.

**Reply by catdogmaus** [reply]
I looked already those templates and that was the reason why to ask about getting it to work. Somebody has already done a similar thing so it is possible. The question is am I able to do this or do I need eventually go back OEM.
I have no idea what it takes to accomplish that for BHT-6000

**Reply by sfromis** [reply]
I've carefully avoided getting into that TuyaMCU stuff with Tasmota, but possibly useful:
https://tasmota.github.io/docs/TuyaMCU/
https://github.com/sillyfrog/Tasmota-Tuya-Helper

**Reply by catdogmaus** [reply]
> I've carefully avoided getting into that TuyaMCU stuff with Tasmota, but possibly useful: https://tasmota.github.io/docs/TuyaMCU/ https://github.com/sillyfrog/Tasmota-Tuya-Helper

These links were of great use, but honestly, they also confused me a little bit. Here is actually a little talk about the Beca thermostat https://tasmota.github.io/docs/TuyaMCU/#becathermostat
It says that for example dpid4 function as mode and dpid102 function as floor temperature but I have not found any option to set those functions in Tasmota. The only things that I have are 72:temperature set and 71:temperature sensor. Also, my bht-6000 needs temperature correction. I did use a little bit this https://templates.blakadder.com/Moes_WHT-HY609.html as a reference point bcs it's using also tuya mcu. It says below in known dpids that 109 is for temp calibration but I have no idea how to implement this. 
It would be great if u or somebody else could help out.

**Comment by yvesdm3000** [comment]
Tuya thermostat generally needs a custom build to enable tuya_time. This enables the time sync function. And then just map the dpIds to the correct function and you should be all set. If the temperature is off by 10 you can use TempRes to change it.  If the offset is anything else , I have a branch with an added command to fix that.

**Reply by catdogmaus** [reply]
Thanks for your answer. All this seems too much hassle for me. I probably will go back to OEM. tuya-local or similar will do.

**Reply by yvesdm3000** [reply]
You can also try one of these binaries. It has even a preset for tuya-thermostat

https://github.com/yvesdm3000/TasmotaFirmwares

**Reply by catdogmaus** [reply]
> You can also try one of these binaries. It has even a preset for tuya-thermostat
> 
> https://github.com/yvesdm3000/TasmotaFirmwares

I did try out bht-002. This did fix the temperature issue....... I think..... Did not measure exactly but result seems more realistic. Unfortunately, this does not fix Tasmota general shortcomings with Thermostats. I cannot use floor temperature bcs Tasmota has only one fnid for temperature sensors, also I can not set fnid for mode switching bcs Tasmota does not have that. After all this in HA I can not put it to the thermostat bcs HA does not recognize it as a thermostat. In HA I have simply a collection of unrelated switches and temperature sensors when actually Thermostat entity is needed. I believe all this is also the reason why the documentation for the Beca thermostat on the official Tasmota help page has not gone anywhere, Tasmota in its current form is simply not suitable for the thermostat. But thanks for effort.

**Reply by yvesdm3000** [reply]
That's correct, a manual config in ha is needed. That's why I started a preset for tuya_thermostat to fix the interface to ha but I didn't find any documentation to implement the discovery bits and I settled with a manual config.

Yves

**Reply by yvesdm3000** [reply]
Some clarification (possibly for other users), it is normal there is only one fnId for temperature since a temperature is a temperature and it is the automation that decides what to do with that. The same with the enum values for modes. The only thing I see lacking (at least for the Thermostats that I use, I don't own a BHT-6000 so I'm not sure what other features it has) is the HA discovery that this is a thermostat. For my use cases, I only use activate/deactivate and occasionally set the target temperature.
